overlord/configstate/configcore/services_test.go

```diff
@@ -337,3 +337,139 @@ func (s *servicesSuite) TestFilesystemOnlyApply(c *C) {
 		{"--root", tmpDir, "mask", "rsyslog.service"},
 	})
 }
+
+func (s *servicesSuite) TestConfigureNetworkSSHListenAddressFailsOnNonCore20(c *C) {
+	err := configcore.Run(coreDev, &mockConf{
+		state: s.state,
+		changes: map[string]interface{}{
+			"service.ssh.listen-address": ":8022",
+		},
+	})
+	c.Assert(err, ErrorMatches, "cannot set ssh listen address configuration on systems older than UC20")
+}
+
+func (s *servicesSuite) TestConfigureNetworkSSHListenAdressFailsWrongRange(c *C) {
+	for _, invalidPort := range []int{0, 65536, -1, 99999} {
+		err := configcore.Run(coreDev, &mockConf{
+			state: s.state,
+			changes: map[string]interface{}{
+				"service.ssh.listen-address": fmt.Sprintf(":%v", invalidPort),
+			},
+		})
+		c.Check(err, ErrorMatches, fmt.Sprintf("cannot validate ssh configuration: cannot use port %v: must be in the range 1-65535", invalidPort))
```

Comment:
nitpick: maybe we can reformulate the the inner errors to avoid "cannot...cannot..." ?

Reply:
Thanks, excellent point. I updated the code and the tests to be more composable.